### ISSUE TITLE
ted_talks_iwslt | Error: Config name is missing

### ISSUE THREAD

**Issue by rayliuca** [huggingface/datasets#6533]
### Describe the bug

Running load_dataset using the newest `datasets` library like below on the ted_talks_iwslt using year pair data will throw an error "Config name is missing"

see also:
https://huggingface.co/datasets/ted_talks_iwslt/discussions/3

likely caused by #6493, where the `and not config_kwargs` part in the if logic was removed 
 https://github.com/huggingface/datasets/blob/ef3b5dd3633995c95d77f35fb17f89ff44990bc4/src/datasets/builder.py#L512

### Steps to reproduce the bug

run: 
```python
load_dataset("ted_talks_iwslt", language_pair=("ja", "en"), year="2015")
```

### Expected behavior

Load the data without error

### Environment info

datasets 2.16.0

**Comment by lhoestq** [comment]
Hi ! Thanks for reporting. I opened https://github.com/huggingface/datasets/pull/6544 to fix this

**Comment by lhoestq** [comment]
We just released 2.16.1 with a fix:

```
pip install -U datasets
```
